cmd/ingress-operator/main.go

```diff
@@ -86,18 +91,21 @@ func main() {
 		os.Exit(1)
 	}
 
-	// TODO: This can be replaced by cluster API when
-	// https://github.com/openshift/installer/pull/1725 is available.
-	clusterConfig := &corev1.ConfigMap{}
-	err = kubeClient.Get(context.TODO(), types.NamespacedName{Namespace: "kube-system", Name: "cluster-config-v1"}, clusterConfig)
-	if err != nil {
-		log.Error(err, "failed to get configmap 'kube-system/cluster-config-v1'")
-		os.Exit(1)
+	// If the cluster proxy config specifies proxy info, create an HTTP client using the proxy
+	// which can be used to configure the HTTP client for all non-apiserver connections.
+	proxyConfig := &configv1.Proxy{}
+	if err := kubeClient.Get(context.TODO(), types.NamespacedName{Name: "cluster"}, proxyConfig); err != nil {
+		if errors.IsNotFound(err) {
```

Comment:
A configv1.Proxy with nil spec/status should exist even if a proxy is not required for 4.2 clusters. With that said, should `os.Exit(1)` be called here?

Reply:
Will it exist in 4.1 clusters?

Reply:
It will not exist in 4.1 clusters. Will this version of the ingress-operator be run in 4.1 clusters?